File: frontend/src/services/playerService.ts
import api from './api'
import { PlayHistory, PageResponse, ApiResponse, PlayProgressRequest } from '@/types'

export const playerService = {
  saveProgress: async (request: PlayProgressRequest): Promise<void> => {
    await api.post('/player/progress', request)
  },

  getProgress: async (videoId: number): Promise<{ videoId: number; progress: number; lastPlayedAt: string }> => {
    const response = await api.get<ApiResponse<{ videoId: number; progress: number; lastPlayedAt: string }>>(
      `/player/progress/${videoId}`
    )
    return response.data.data
  },

  getHistory: async (params: { page?: number; size?: number } = {}): Promise<PageResponse<PlayHistory>> => {
    const response = await api.get<ApiResponse<PageResponse<PlayHistory>>>('/player/history', { params })
    return response.data.data
  },

  clearHistory: async (): Promise<void> => {
    await api.delete('/player/history')
  },
}

File: backend/src/main/java/com/solo/video/service/PlayerService.java
package com.solo.video.service;

import com.solo.video.dto.request.PlayProgressRequest;
import com.solo.video.dto.response.PlayHistoryResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface PlayerService {
    
    void saveProgress(PlayProgressRequest request);
    
    PlayHistoryResponse getProgress(Long videoId);
    
    Page<PlayHistoryResponse> getPlayHistory(Pageable pageable);
    
    void clearPlayHistory();
    
    void incrementPlayCount(Long videoId);
}
